p2p/src/swarm/tests/peerdb.rs

```diff
@@ -0,0 +1,559 @@
+
+    peer_id
+}
+
+fn add_banned_peer(peerdb: &mut PeerDb<Libp2pService>) -> PeerId {
+    let (id, info) = make_peer_info();
+    peerdb.register_peer_info(Multiaddr::empty(), info);
+    peerdb.ban_peer(&id);
+
+    id
+}
+
+#[test]
+fn num_active_peers() {
+    let mut peerdb = PeerDb::<Libp2pService>::new(Arc::new(config::P2pConfig::new()));
+
+    assert_eq!(peerdb.idle_peer_count(), 0);
+    assert_eq!(peerdb.active_peer_count(), 0);
+
+    // add three active peers
+    for _ in 0..3 {
+        let _id = add_active_peer(&mut peerdb);
+    }
+    assert_eq!(peerdb.idle_peer_count(), 0);
+    assert_eq!(peerdb.active_peer_count(), 3);
+    assert_eq!(peerdb.peers().len(), 3);
+
+    // add 2 idle peers
+    for _ in 0..2 {
+        let _id = add_idle_peer(&mut peerdb);
+    }
+    assert_eq!(peerdb.idle_peer_count(), 2);
+    assert_eq!(peerdb.active_peer_count(), 3);
+    assert_eq!(peerdb.peers().len(), 5);
+
+    // add 4 discovered peers
+    for _ in 0..2 {
+        let _id = add_discovered_peer(&mut peerdb);
+    }
+    assert_eq!(peerdb.idle_peer_count(), 4);
+    assert_eq!(peerdb.active_peer_count(), 3);
+    assert_eq!(peerdb.peers().len(), 7);
+
+    // add 5 banned peesrs
+    for _ in 0..5 {
+        let _id = add_banned_peer(&mut peerdb);
+    }
+    assert_eq!(peerdb.idle_peer_count(), 4);
+    assert_eq!(peerdb.active_peer_count(), 3);
+    assert_eq!(peerdb.peers().len(), 12);
+}
+
+#[test]
+fn is_active_peer() {
+    let mut peerdb = PeerDb::<Libp2pService>::new(Arc::new(config::P2pConfig::new()));
+
+    let id1 = add_active_peer(&mut peerdb);
+    assert!(peerdb.is_active_peer(&id1));
+
+    let id2 = add_idle_peer(&mut peerdb);
+    assert!(!peerdb.is_active_peer(&id2));
+
+    let id3 = add_discovered_peer(&mut peerdb);
+    assert!(!peerdb.is_active_peer(&id3));
+
+    let id4 = add_banned_peer(&mut peerdb);
+    assert!(!peerdb.is_active_peer(&id4));
+}
+
+#[test]
+fn adjust_peer_score() {
+    // peer banned after adjustment
+    {
+        let mut peerdb = PeerDb::<Libp2pService>::new(Arc::new(config::P2pConfig::new()));
+
+        let id = add_active_peer(&mut peerdb);
+        assert!(peerdb.adjust_peer_score(&id, 100));
+        assert!(peerdb.banned().contains(&id));
+    }
+
+    // higher threshold, no ban
+    {
+        let mut config = config::P2pConfig::new();
+        config.ban_threshold = 200;
+        let mut peerdb = PeerDb::<Libp2pService>::new(Arc::new(config));
+
+        let id = add_active_peer(&mut peerdb);
+        assert!(!peerdb.adjust_peer_score(&id, 100));
+        assert!(!peerdb.banned().contains(&id));
+    }
+
+    // lower threshold, ban for more minor offense
+    {
+        let mut config = config::P2pConfig::new();
+        config.ban_threshold = 20;
+        let mut peerdb = PeerDb::<Libp2pService>::new(Arc::new(config));
+
+        let id = add_active_peer(&mut peerdb);
+        assert!(peerdb.adjust_peer_score(&id, 30));
+        assert!(peerdb.banned().contains(&id));
+    }
```

Comment:
I'd like to see these test cases each having its own test function. Makes it easier to see which fails and re-run them on individual basis. I appreciate coming up with names is annoying but you already have comments there to take inspiration from.